### ISSUE TITLE
Using color map for vertex colored triangles

### ISSUE THREAD

**Issue by bhairav-thakkar** [helix-toolkit/helix-toolkit#1689]
I am using the Helixtoolkit/wpf/sharpdx. 

What I have is a finite element mesh with triangles having different values over the domain. At the moment, I am shading the mesh by giving a color at each vertex related to the value at each vertex. It works pretty well.

Now, I want to implement a color map - say Jet, which has 7 different colors mapping with the value. It would work if the mesh is sufficiently fine. However, for course meshes, when the triangle size is large, colors jump on the color map scale, obviously because the triangle is shaded by 2D linear interpolation.

Is there a way by which we can specify a color map so that when triangle vertices are assigned a color, the domain colors are picked up from the map instead of 2D linear interpolation?

I wish to achieve something like this:

![image](https://user-images.githubusercontent.com/32364304/145248638-9669fd67-8e70-4aef-a31c-d0a931276366.png)

Thanks
Bhairav

**Comment by holance** [comment]
Yes, you need to use texture coordinates (color coordinates according to the gradient map) combine with a diffuse map (the gradient map you generated).

**Comment by bhairav-thakkar** [comment]
That sounds great.. Is there any example that does something similar to this?

Thanks

**Comment by holance** [comment]
Probably you can refer to color stripe material
https://github.com/helix-toolkit/helix-toolkit/blob/3d353b2ad7d22b421d893e44383f6eac7289c1b6/Source/Examples/WPF.SharpDX/MaterialDemo/MainViewModel.cs#L91
